yjit_iface.c

```diff
@@ -747,6 +760,23 @@ get_stat_counters(rb_execution_context_t *ec, VALUE self)
             name_reader = name_end;
         }
     }
+
+    {
+        // For each entry in exit_op_count, add a stats entry with key "exit_INSTRUCTION_NAME",
+        // where the value is the count of side exits for that instruction.
+
+        char key_string[rb_vm_max_insn_name_size + 6]; // Leave room for "exit_" and a final NUL
+        strncpy(key_string, "exit_", 6);               // Copy 6 characters to include the NUL.
+        for (int i = 0; i < VM_INSTRUCTION_SIZE; i++) {
+            const char *i_name = insn_name(i); // Look up Ruby's NUL-terminated insn name string
+            strncpy(key_string + 5, i_name, rb_vm_max_insn_name_size + 1);
```

Comment:
This looks right but might be simpler as `snprintf`, to avoid counting lengths by hand.

Reply:
That's a good idea. Done.